### PR TITLE
Add Vector-via-cql tests for Coordinator

### LINKED ISSUE #2698: Add CQL tests for `dse-next` backend to verify basic Vector functionality over cql
Now that we have "dse-next" backend with vector search capabilities, let's add simple ITs for verifying that basic vector functionality works over cql:

* Ability to create tables with vector column(s)
* Ability to insert rows with vector values
* Ability to perform vector search

Initial version should work more as smoke test / sanity check with just simple verification. More test cases may be added over time to cover regressions and such.
(that is, we are not validating full vector functionality but rather that basic functionality is exposed via cql).

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds basic tests to verify Vector functionality access using CQL for dse-next backend

**Which issue(s) this PR fixes**:
Fixes #2698

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
As expected, passes on `dse-next`, fails on other backends, due to test that runs against all backends currently. Need to figure out how to check for vector support (lack thereof) to skip tests on such backends.

File: coordinator/pom.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
     <!-- Release -->
     <project.scm.id>github</project.scm.id>
     <doclint>none</doclint>
-    <driver.version>4.16.0</driver.version>
+    <driver.version>4.17.0</driver.version>
```
Comment (tatu-at-datastax) [inline coordinator/pom.xml:58]: Needed to get access to vector-aware Result/Row accessors.

File: coordinator/testing/pom.xml [hunk 2]
```diff
@@ -365,6 +365,7 @@
                     <ccm.version>4.0.10</ccm.version>
                     <stargate.test.nodes>1</stargate.test.nodes>
                     <stargate.test.persistence.module>persistence-cassandra-4.0</stargate.test.persistence.module>
+                    <stargate.test.backend.vsearch>false</stargate.test.backend.vsearch>
```
Comment (tatu-at-datastax) [inline coordinator/testing/pom.xml:368]: No clean way to auto-detect via driver or ccm so just add another system property for tests

File: coordinator/persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java [hunk 3]
```diff
@@ -192,6 +192,11 @@ public int testsSkipped() {
     public int errorsDetected() {
       return errorsInTest.get();
     }
+
+    @Override
+    public boolean supportsVSearch() {
+      return Boolean.getBoolean("stargate.test.backend.vsearch");
```
Comment (jeffreyscarpenter) [inline coordinator/persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java:198]: defaults to false?

**Review by jeffreyscarpenter — APPROVED** [review]
Great addition!

File: coordinator/persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java (hunk 3 above)
Comment (tatu-at-datastax) [reply coordinator/persistence-test/src/main/java/io/stargate/it/storage/ExternalStorage.java:198]: Yes.

**Pull request merged**